### PR TITLE
feat(Rest): New end points for project/component/release usage summary

### PR DESCRIPTION
**Pull request by nutanv1contr**
Created three rest end points like

* http://sw360.com/resource/api/projects/usedBy/b7b07cadabb084a68c746b0c0705581d
* http://sw360.com/resource/api/components/usedBy/b7b07cadabb084a68c746b0c07007b99
* http://sw360.com/resource/api/releases/usedBy/ac8a53df29af6391bd76f61aea001336

which will display a short detail of the project/component/release usage summary along with the hyperlink to navigate to the resource . 

image

closes : #704

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java [hunk 1]
```diff
@@ -60,6 +60,23 @@ public Project getProjectForUserById(String projectId, User sw360User) throws TE
         return sw360ProjectClient.getProjectById(projectId, sw360User);
     }
 
+    public Set<Project> getUsedByProjectsForUserByProjectId(String projectId, User sw360User) throws TException {
```
[the remark this answers is not in the document]
Comment (nutanv1contr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java]: Done

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java [hunk 2]
```diff
@@ -60,6 +60,23 @@ public Project getProjectForUserById(String projectId, User sw360User) throws TE
...
+        ProjectService.Iface sw360ProjectClient = getThriftProjectClient();
+        return sw360ProjectClient.searchLinkingProjects(projectId, sw360User);
+    }
+
+    public Set<Project> getUsedByProjectsByReleaseIds(Set<String> releaseids, User sw360User)
```
[the remark this answers is not in the document]
Comment (nutanv1contr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java]: Done

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release/Sw360ReleaseService.java [hunk 3]
```diff
@@ -96,6 +103,17 @@ public RequestStatus deleteRelease(String releaseId, User sw360User) throws TExc
         return sw360ComponentClient.deleteRelease(releaseId, sw360User);
     }
 
+    public Set<Project> getUsedByProjectsForUserById(String releaseId, User sw360User) throws TException {
```
[the remark this answers is not in the document]
Comment (nutanv1contr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release/Sw360ReleaseService.java]: Done

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/Sw360ComponentService.java [hunk 4]
```diff
@@ -56,6 +63,23 @@ public Component getComponentForUserById(String componentId, User sw360User) thr
         return sw360ComponentClient.getComponentById(componentId, sw360User);
     }
 
+    public Set<Project> getUsedByProjectsForUserById(String componentId, User sw360User) throws TException {
```
[the remark this answers is not in the document]
Comment (nutanv1contr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/Sw360ComponentService.java]: Done

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/Sw360ComponentService.java [hunk 5]
```diff
@@ -56,6 +63,23 @@ public Component getComponentForUserById(String componentId, User sw360User) thr
...
+        ComponentService.Iface sw360ComponentClient = getThriftComponentClient();
+        Component component = sw360ComponentClient.getComponentById(componentId, sw360User);
+        Set<String> releaseIds = SW360Utils.getReleaseIds(component.getReleases());
+        Set<Project> usedByProjects = projectService.getUsedByProjectsByReleaseIds(releaseIds, sw360User);
+
+        return usedByProjects;
+    }
+
+    public Set<Component> getUsedByComponentsForUserById(String componentId, User sw360User) throws TException {
+        ComponentService.Iface sw360ComponentClient = getThriftComponentClient();
```
[the remark this answers is not in the document]
Comment (nutanv1contr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/Sw360ComponentService.java]: Made the necessary changes

**Comment by akapti** [comment]
Tested the changes, working fine. 
verified the documentation.
code looks good.

**Pull request merged**